templates/secret.yaml

```diff
@@ -0,0 +1,26 @@
+{{- if not .Values.security.endpointSecretName }}
+  {{- if or (.Release.IsInstall) (.Values.security.endpointSecretPassword) }}
+{{- $password := default (.Values.security.endpointSecretPassword) (randAlphaNum 8) }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ printf "%s-generated-secret" (include "infinispan-helm-charts.name" .) }}
+  labels:
+    app: infinispan-secret-identities
+  {{- include "infinispan-helm-charts.labels" . | nindent 4 }}
+  annotations:
+    "helm.sh/resource-policy": no-update
+type: Opaque
+stringData:
+  username: developer
+  password: {{ $password }}
+  identities.yaml: |-
```

Comment:
Yes, good point, however I thought about to add ServiceMonitor also and we need static username and password as key/value parameters. We can repeat some operator logic and admin-identties for that. What do you think?

Reply:
> ```yaml
> password: {{ randAlphaNum 8 }}
> ```

Seems to not possible use functions inside the values.yaml files. I saw long histories in helm GitHub but without solutions at all.

Reply:
Yes, but `.Value.identities.credentials` should be always list with static values. How we will generate random password if needed.

Reply:
> Seems to not possible use functions inside the values.yaml files. I saw long histories in helm GitHub but without solutions at all.

I hadn't read this when I replied :sweat_smile:

Reply:
We can add an identities structure optional and if customer will provide it, propagate to the Secret, otherwise use an auto-generated password and remove endpointSecretPassword field from the structure

Reply:
Sounds good. Another consideration, which again can come after this PR, would be to replace the identities yaml with raw `user.properties` and `group.properties` files that can be handwritten or created via the Infinispan CLI.

Reply:
Hmm I realised a problem with using `credentials[0]` now that Infinispan has Authorization enabled by default (coming to the images very soon). If the first credential specified in the identities file does not have the appropriate role to allow metrics to be gathered, then the ServiceMonitor will fail.

I think we should do a couple of things.

For the generated credentials use-case, we can simply generate the password and create a identities.yaml with a single user, which is what you already have.

For user provided identities.yaml, the user should explicitly configure a username/password value for the ServiceMonitor. For example, this could be `monitoring.username` and `monitoring.password`. This has the added benefit of being future proof when we move towards providing `user.properties` and `groups.properties` files instead of identities.yaml.

Reply:
I've extracted first username/password pair the service monitor as separate keys in the Secret.